### PR TITLE
#27500 Prepare Search module Tests for PHPUnit 8

### PR DESCRIPTION
**Pull request by lbajsarowicz**
### Description (*)
Preparing PHPUnit Tests to run with PHPUnit 8.
PRs for each module is separated to make tests easier to merge.

### Fixed Issues (if relevant)
1. magento/magento2#27500

### Manual testing scenarios (*)
N/A

### Questions or comments
PR #27701 needs to be merged first (otherwise PHPUnit tests fail)

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Search/Test/Unit/Model/SearchEngine/ConfigTest.php [hunk 1]
```diff
@@ -1,41 +1,46 @@
...
     protected $dataStorageMock;
 
-    /** @var \Magento\Framework\TestFramework\Unit\Helper\ObjectManager */
+    /** @var ObjectManager */
     protected $objectManager;
 
-    protected function setUp()
+    protected function setUp(): void
     {
-        $this->dataStorage = $this->createMock(\Magento\Search\Model\SearchEngine\Config\Data::class);
```
Comment (dmytro-ch) [inline app/code/Magento/Search/Test/Unit/Model/SearchEngine/ConfigTest.php:18]: Most likely the test fails due to the missing `dataStorageMock`.

**Review by dmytro-ch — CHANGES_REQUESTED** [review]

File: app/code/Magento/Search/Test/Unit/Model/SearchEngine/ConfigTest.php (hunk 1 above)
Comment (lbajsarowicz) [reply app/code/Magento/Search/Test/Unit/Model/SearchEngine/ConfigTest.php:18]: PHPStan should detect that. But if not - I'm going to fix that either way. 
Thank you!

**Review by dmytro-ch — APPROVED** [review]
Fixed the tests failures. The PR ready to be merged. ✔️

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-7448](https://jira.corp.magento.com/browse/ENGCOM-7448) has been created to process this Pull Request

**Pull request merged**